src/main/kotlin/com/cognifide/gradle/aem/tooling/tail/Tailer.kt

```diff
@@ -0,0 +1,44 @@
+package com.cognifide.gradle.aem.tooling.tail
+
+import java.io.BufferedReader
+
+class Tailer(private val source: LogSource, private val destination: LogDestination, val name: String = "") {
+
+    private val parser = Parser()
+
+    private var lastLogChecksum = ""
+
+    fun tail() {
+        val logs = source.processChunk { chunkReader ->
+            parser.parseLogs(chunkReader)
+        }
+        val newLogs = determineNewLogs(logs)
+
+        if (newLogs.isNotEmpty()) {
+            lastLogChecksum = newLogs.last().checksum
+            destination.dump(newLogs)
+        }
+    }
+
+    private fun determineNewLogs(logs: List<Log>): List<Log> {
+        val reversedLogs = logs.reversed()
+        val lastFetchedLog = reversedLogs.find { log -> lastLogChecksum == log.checksum }
+        return when {
+            lastFetchedLog == null -> logs
+            logs.last() === lastFetchedLog -> emptyList()
+            else -> logs.slice(logs.indexOf(lastFetchedLog) + 1 until logs.size)
+        }
+    }
+}
+
+interface LogDestination {
```

Comment:
This whole tool is rather closed and final. I am not sure if interfaces and any abstraction introduced is a good way to go

Reply:
Thanks to interfaces I can test it easily in memory